velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -209,4 +209,21 @@ struct LastDayFunction {
   }
 };
 
+template <typename T>
+struct DateSubFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Date>& result,
+      const arg_type<Date>& date,
+      const int32_t value) {
+    int64_t subValue = 0 - (int64_t)value;
+    if (subValue != (int32_t)subValue) {
+      VELOX_UNSUPPORTED("integer overflow");
```

Comment:
Does this behavior match Spark?

Reply:
Before implementation is wrong, treat kMin ( -2147483648) as overflow, this didn't match spark.
Update the function implementation to judge the corner case when input is kMin.
Kindly help to review. Thanks.